grakn-graql/src/main/java/ai/grakn/graql/internal/hal/HALConceptData.java

```diff
@@ -243,13 +234,15 @@ private void attachRelation(Representation halResource, Concept rel, TypeName ro
 
     private void generateRelationEmbedded(Representation halResource, Relation rel, int separationDegree) {
 
-        rel.rolePlayers().forEach((roleType, instance) -> {
-            if (instance != null) {
-                Representation roleResource = factory.newRepresentation(resourceLinkPrefix + instance.getId() + this.keyspace)
-                        .withProperty(DIRECTION_PROPERTY, OUTBOUND_EDGE);
-                handleConcept(roleResource, instance, separationDegree - 1);
-                halResource.withRepresentation(roleType.getName().getValue(), roleResource);
-            }
+        rel.allRolePlayers().forEach((roleType, instanceSet) -> {
+            instanceSet.forEach(instance -> {
+                if (instance != null) {
+                    Representation roleResource = factory.newRepresentation(resourceLinkPrefix + instance.getId() + this.keyspace)
+                            .withProperty(DIRECTION_PROPERTY, OUTBOUND_EDGE);
+                    handleConcept(roleResource, instance, separationDegree - 1);
+                    halResource.withRepresentation(roleType.getName().getValue(), roleResource);
+                }
+            });
```

Comment:
Dashboard will be broken. I say we merge and then ask Marco to take a look at it. This PR is blocking on a few things.